### PR TITLE
Batch Handler V8-lane bug

### PR DESCRIPTION
**Pull request by lamai93**
When handling a batch request, queue each job with priority given by its lane instead of using the slowest lane for all.
Make the Batch handler aware of Handler that do not return a result immediately.

- [x] Execute each handler with its own priority.
- [x] Support waiting rest handlers.
- [ ] Execute all handlers in parallel.
- [ ] Clean up

### CODE REVIEW

**Comment by fceller** [comment]
@lamai93 LGTM

**Comment by lamai93** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/694/

File: arangod/RestHandler/RestBatchHandler.cpp [hunk 1]
```diff
@@ -66,214 +67,240 @@ RestStatus RestBatchHandler::executeVst() {
   return RestStatus::DONE;
 }
 
-RestStatus RestBatchHandler::executeHttp() {
+void RestBatchHandler::processSubHandlerResult(std::shared_ptr<RestHandler> handler)
```
Comment (jsteemann) [inline arangod/RestHandler/RestBatchHandler.cpp]: The `shared_ptr` is copied here, but there is no need to copy it. I suggest to change the signature to use a const reference instead.

File: arangod/RestHandler/RestBatchHandler.h [hunk 2]
```diff
@@ -37,9 +37,10 @@ struct MultipartMessage {
         boundaryLength(boundaryLength),
         messageStart(messageStart),
         messageEnd(messageEnd){};
+  MultipartMessage(){}
```
Comment (jsteemann) [inline arangod/RestHandler/RestBatchHandler.h]: This is not initializing any member and should be avoided.

File: arangod/RestHandler/RestBatchHandler.h [hunk 3]
```diff
@@ -81,6 +82,15 @@ class RestBatchHandler : public RestVocbaseBaseHandler {
...
   // extract the next part from a multipart message
   bool extractPart(SearchHelper*);
+
+private:
+  bool executeNextHandler();
+  void processSubHandlerResult(std::shared_ptr<RestHandler> handler);
+
+  MultipartMessage _multipartMessage;
+  SearchHelper _helper;
+  size_t _errors;
```
Comment (jsteemann) [inline arangod/RestHandler/RestBatchHandler.h]: `_errors` should be initialized in the initializer list of RestBatchHandler, but not in `executeHttp()`

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/726/

**Pull request merged**